### PR TITLE
Adds initial mhv-supply-reordering application

### PR DESCRIPTION
**Pull request by radavis**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Adds the initial mhv-supply-reordering application to the platform
- https://github.com/department-of-veterans-affairs/content-build/pull/2354 Sets `vagovprod: false` so that app is only available in lower environments.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#97137
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/96513

## Testing done

- Follow readme instructions to develop the app, locally
- UI for alerting conditions: see `containers/Alerts.jsx` and `components/Alert*.jsx`
- UI for displaying and selecting supplies, editing email and address
- Note: Form submission won't work because this application is a work in progress.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-supply-reordering/components/SuppliesUnavailable.jsx
Comment (acrollet) [inline src/applications/mhv-supply-reordering/components/SuppliesUnavailable.jsx:10]: It could be nice to move this logic to a `sortedSupplies` utility function

File: src/applications/mhv-supply-reordering/components/alerts/AlertSomethingWentWrong.jsx
Comment (acrollet) [inline src/applications/mhv-supply-reordering/components/alerts/AlertSomethingWentWrong.jsx:14]: Should this be a heading? Genuinely not sure.

File: src/applications/mhv-supply-reordering/config/form.js
Comment (acrollet) [inline src/applications/mhv-supply-reordering/config/form.js:97]: Worth keeping this?

File: src/applications/mhv-supply-reordering/mocks/feature-toggles/index.js
Comment (acrollet) [inline src/applications/mhv-supply-reordering/mocks/feature-toggles/index.js:1]: This is a neat approach but I don't love the lodash dependency since it's pretty heavy. (maybe it doesn't matter if it gets tree shaken?)

File: src/applications/mhv-supply-reordering/pages/chooseSupplies.jsx
Comment (acrollet) [inline src/applications/mhv-supply-reordering/pages/chooseSupplies.jsx:11]: maybe we could move this to a shared utility and use it other places the pattern occurs? (I know I saw it at least one other place)

File: src/applications/mhv-supply-reordering/selectors/index.js
Comment (acrollet) [inline src/applications/mhv-supply-reordering/selectors/index.js:7]: do we have unit tests for these selectors?

**Review by acrollet — COMMENTED** [review]
This is excellent work! I really love seeing all the decomposition and how much cleaner everything is. I have a few small comments but don't see anything show-stopping at all.

File: src/applications/mhv-supply-reordering/components/EditAddress.jsx
Comment (acrollet) [reply src/applications/mhv-supply-reordering/components/EditAddress.jsx:33]: sorry if I jumped the gun on the review!

File: src/applications/mhv-supply-reordering/components/EditEmail.jsx
Comment (radavis) [reply src/applications/mhv-supply-reordering/components/EditEmail.jsx:4]: Hmmm.... Not yet, but we should!

File: src/applications/mhv-supply-reordering/components/alerts/AlertSomethingWentWrong.jsx
Comment (radavis) [reply src/applications/mhv-supply-reordering/components/alerts/AlertSomethingWentWrong.jsx:14]: Couldn't hurt to make this an h4, but I might hold off changing it until an a11y review.

File: src/applications/mhv-supply-reordering/config/form.js
Comment (radavis) [reply src/applications/mhv-supply-reordering/config/form.js]: Gone!

**Review by nichia — COMMENTED** [review]
The structure/setup for supply reordering is looking good.

**Review by dcloud — APPROVED** [review]
I think this is good to move forward. Just some minor feedback below.

One piece of feedback is that you might consider moving/merging your mocks into `src/platform/mhv/api/mocks`. This is something I keep meaning to discuss around the mhv-landing-page mocks as well, since it seems potentially useful to buy into the shared MHV mocks.

I also approve this assuming you'll address the proptypes warnings in a future PR, though I am ambivalent on the value of proptypes validation myself.

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**
